### ISSUE TITLE
php-parser integrity error after plugin-php update

### ISSUE THREAD

**Issue by jebarjonet** [prettier/plugin-php#1836]
I got an integrity error when I updated `@prettier/plugin-php` from `0.16.1` to `0.17.5`.

```
npm ERR! code EINTEGRITY
npm ERR! sha512-zhM2ggKlHIj4BhpSzoj4zQWAOG7yi/pWdcIWPvyYwl2Na9BNpddbQCU/WBcZCwBZKT1rFbEYkeBXC1gmHZIY8Q== integrity checksum failed when using sha512: wanted sha512-zhM2ggKlHIj4BhpSzoj4zQWAOG7yi/pWdcIWPvyYwl2Na9BNpddbQCU/WBcZCwBZKT1rFbEYkeBXC1gmHZIY8Q== but got sha512-T4ypA7AQdplhTSawCk1jw4oIBF6ytumfagzmHyZpP8te0LF2kizXlrVKWfVyEf94cyTmO7QOAH73fhyiBk0eRQ==. (144420 bytes)
```

It is the same issue as this one https://github.com/prettier/plugin-php/issues/1813 but I am not satisfied with the answer. What happens is:
- I use `@prettier/plugin-php` `0.16.1`, it installs the package `php-parser` `3.0.2` release from npm
- I update `@prettier/plugin-php` to `0.17.5`, [this update](https://github.com/prettier/plugin-php/pull/1656/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R16) and many others have set a Github url as the version to use instead of npm registry for `php-parser`, but still a `3.0.2` release
- when running `npm i`, and even after deleting `package-lock.json`, I get the above error

That's because `npm` builds a `package-lock.json` with the new version's hash from Github version of `php-parser`. Then it tries to install packages, but there is a mismatch between the hash of the `package-lock` version's (from [Github](https://github.com/glayzzle/php-parser/blob/4c5b0675f52c0baab2e5b10a4e50e5d7a79b2767/package.json)) and the one already in `node-modules` (from npm registry), both `3.0.2` release.

You need to delete `node-modules` to reinstall everything without mismatching cache.

Can you please use an official npm release of `php-parser` instead of a Github custom version? It would prevent hashes comparisons from cached `node-modules` mismatches !

**Comment by hexsprite** [comment]
Any way to get around this? I am using the latest plugin-php and still getting this issue

**Comment by watsonhaw5566** [comment]
> 有什么办法可以解决这个问题？我正在使用最新的 plugin-php 并且仍然遇到这个问题


I'm also too.Alway Stop in install `p-defer` package, Then throw 443

![image](https://user-images.githubusercontent.com/4210367/149460101-d56721d0-d07f-42fb-a1cc-7e1bdaa08c59.png)

**Comment by czosel** [comment]
We'll a new release of php-parser to get this fixed, which is currently blocked by https://github.com/glayzzle/php-parser/pull/809.

**Comment by jebarjonet** [comment]
@czosel awesome ! I saw you merged the PR. Are you planning on releasing a new version on npm soon ?

**Comment by czosel** [comment]
@jebarjonet Yep, that’s the plan 🙂

**Comment by czosel** [comment]
Should be fixed in `v0.18.0` :tada:
